plugins/uefi-capsule/fu-uefi-esrt-linux.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2021 3mdeb Embedded Systems Consulting
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-common.h"
+#include "fu-uefi-common.h"
+#include "fu-uefi-esrt.h"
+
+struct _FuUefiEsrtEntry {
+	GObject		 parent_instance;
+	gchar		*path;
+};
+
+G_DEFINE_TYPE (FuUefiEsrtEntry, fu_uefi_esrt_entry, G_TYPE_OBJECT)
+
+struct _FuUefiEsrt {
+	GObject		 parent_instance;
+	GPtrArray	*entries;		/* of FuUefiEsrtEntry */
+};
+
+G_DEFINE_TYPE (FuUefiEsrt, fu_uefi_esrt, G_TYPE_OBJECT)
+
+static FuUefiEsrtEntry *
+fu_uefi_esrt_entry_new (gchar *path)
+{
+	FuUefiEsrtEntry *self = g_object_new (FU_TYPE_UEFI_ESRT_ENTRY, NULL);
+	self->path = path;
+	return FU_UEFI_ESRT_ENTRY (self);
+}
+
+static void
+fu_uefi_esrt_entry_init (FuUefiEsrtEntry *self)
+{
+	self->path = NULL;
```

Comment:
@hughsie is this actually needed?  Or does it init NULL automatically?

Reply:
It's not needed

Reply:
I removed this, [the docs](https://developer.gnome.org/gobject/stable/gobject-Type-Information.html#GInstanceInitFunc) say (if that's documentation for this function):

    The extended members of instance are guaranteed to have been filled with zeros before this function is called.